src/rpdk/core/contract/resource_client.py

```diff
@@ -117,6 +117,7 @@ def __init__(
     ):  # pylint: disable=too-many-arguments
         self._schema = schema
         self._session = create_sdk_session(region)
+        self._role_arn = role_arn
         self._creds = get_temporary_credentials(
```

Comment:
is this _creds member variable needed if we are calling get_temporary_credentials every time? if not we may just be able to remove it.

Reply:
Good catch. It will be used below to get the account ID once. I will remove it as instance member var.

Reply:
some tests failing from this:
https://github.com/aws-cloudformation/cloudformation-cli/blob/88a4d8762d855e5d9e10609915fe93d41fe21be8/tests/contract/test_resource_client.py#L85
https://github.com/aws-cloudformation/cloudformation-cli/blob/88a4d8762d855e5d9e10609915fe93d41fe21be8/tests/contract/test_resource_client.py#L127

Reply:
Pretty funny, my local `pytest` works fine.

Reply:
OK. realize I need to do `pip3 install .` before the run the `pytest`. Am I doing something wrong?  can I run the `pytest` again the code in local workspace instead of installed package?